src/pages/cats.js

```diff
@@ -177,10 +177,27 @@ const useStyles = makeStyles(() => ({
 }))
 const Cats = ({ pageContext }) => {
   const cx = useStyles()
+  const checkIfReferrered = () => {
+    let isReferred = false
+    if (pageContext && pageContext.referrer) {
+      isReferred = true
+    } else {
+      const paramRefId = parseInt(getUrlParameterValue('r'))
+      if (!isNaN(paramRefId)) {
+        isReferred = true
+      }
+    }
+    const userReferrerId = getUrlParameterValue('u')
+    if (userReferrerId !== null && userReferrerId !== undefined) {
+      isReferred = true
+    }
+    return isReferred
+  }
   const [
     showUnsupportedBrowserMessage,
     setShowUnsupportedBrowserMessage,
   ] = useState(false)
+  const [isReferral] = useState(checkIfReferrered())
```

Comment:
I believe it's best practice to set client-determined state after mount. Here, we'd start with a default state of `false`. Gatsby is rendering page HTML on build (always no referrer), and if the state changes before mount (client side with `u` param), there will be a hydration conflict.